pkg/yurtctl/cmd/convert/edgenode.go

```diff
@@ -0,0 +1,276 @@
+package convert
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/alibaba/openyurt/pkg/projectinfo"
+	enutil "github.com/alibaba/openyurt/pkg/yurtctl/util/edgenode"
+	kubeutil "github.com/alibaba/openyurt/pkg/yurtctl/util/kubernetes"
+	"github.com/alibaba/openyurt/pkg/yurthub/healthchecker"
+)
+
+const (
+	kubeletConfigRegularExpression = "\\-\\-kubeconfig=.*kubelet.conf"
+	apiserverAddrRegularExpression = "server: (http(s)?:\\/\\/)?[\\w][\\w]{0,62}(\\.[\\w][-\\w]{0,62})+(:[\\d]{1,5})?"
+	hubHealthzCheckFrequency       = 10 * time.Second
+	failedRetry                    = 5
+	filemode                       = 0666
+	dirmode                        = 0755
+)
+
+// EdgenodeOptions holds the command-line options for 'convert edgenode' sub command
+type ConvertEdgeNodeOptions struct {
+	clientSet       *kubernetes.Clientset
+	YurthubImage    string
+	JoinToken       string
+	PodMainfestPath string
+	kubeletSvcPath  string
+	openyurtDir     string
+}
+
+func NewConvertEdgeNodeOptions() *ConvertEdgeNodeOptions {
+	return &ConvertEdgeNodeOptions{}
+}
+
+func NewConvertEdgeNodeCmd() *cobra.Command {
+	c := NewConvertEdgeNodeOptions()
+	cmd := &cobra.Command{
+		Use:   "edgenode",
+		Short: "Converts the kubernetes node to a yurt node",
+		Run: func(cmd *cobra.Command, _ []string) {
+			if err := c.Complete(cmd.Flags()); err != nil {
+				klog.Fatalf("fail to complete the convert edgenode option: %s", err)
+			}
+			if err := c.RunConvertEdgeNode(); err != nil {
+				klog.Fatalf("fail to covert the kubernetes node to a yurt node: %s", err)
+			}
+		},
+	}
+
+	cmd.Flags().String("join-token", "", "the join token.")
+	cmd.Flags().String("yurthub-image", "openyurt/yurthub:latest", "the yurthub image.")
+	cmd.Flags().String("pod-manifest-path", "/etc/kubernetes/manifests", "Path to the directory on edge node containing static pod files.")
+
+	return cmd
+}
+
+func (c *ConvertEdgeNodeOptions) Complete(flags *pflag.FlagSet) error {
+	yurthubImage, err := flags.GetString("yurthub-image")
+	if err != nil {
+		return err
+	}
+	c.YurthubImage = yurthubImage
+
+	joinToken, err := flags.GetString("join-token")
+	if err != nil {
+		return err
+	}
+	c.JoinToken = joinToken
+
+	podMainfestPath, err := flags.GetString("pod-manifest-path")
+	if err != nil {
+		return err
+	}
+	c.PodMainfestPath = podMainfestPath
+
+	c.clientSet, err = kubeutil.GenClientSet(flags)
```

Comment:
if there is no kubeconfig in all `flag`, `KUBECONFIG`, `~/.kube/config`, it will return err.

@rambohe-ch , how about use `/etc/kubernetes/kubelet.conf` if err occurs? `kubelet.conf` has enougth authority to set label for current edge node.

Reply:
+1 for `using /etc/kubernetes/kubelet.conf` as the default.

Reply:
@GsssC how about use `InClusterConfig` at the edge node? plus add a serviceaccount and rabc for servant pod.

Reply:
`yurtctl convert/revert edgenode` could be used directly on the single node. In this case, we can't use `InClusterConfig` at the edge node because it is not running in a pod environment. Maybe using `/etc/kubernetes/kubelet.conf` as the default is better?